### DISCUSSION TITLE
[bug] Metadata menu creates memory leak when click on fileClass icon in file view

### DISCUSSION THREAD

**Discussion by derrickwheals** [mdelobelle/metadatamenu#381] [General]
For the last few days, metadata menu caused Obsidian's memory usage to rocket up from sub 1Gb to 4 to 5Gb till Obsidian crashes. It appears to happen when I click on the metadata menu icon on a fileClass file to edit the fields.

I am running Obsidian : v1.4.2 (beta) and Metadata Menu v0.5.1

**Comment by technotherion** [comment]
This happened to me recently as well, and it had to do with formulas recalculating and/or updating in Table View infinitely until Obsidian crashes. 
Here's some things that I have found that help:
1. Back up your vault. When this happened to me, quite a few pages randomly became blank when Obsidian crashed. 
2. Turn off any fields that auto-update, especially if they inherit from other auto-updating fields. 
3. in fileClass settings, set "max records per page" to 1. You can do this with the plugin disabled, and then enable the plugin to test. 
I found that upon restarting Obsidian, my memory and CPU usage decreased drastically after making these changes.

**Comment by seannik** [comment]
Confirm this behaviour.  v1.3.7 and 1.4.2.
Identified Metadata Menu through process of elimination, as cause.
Disabling formula auto-update resolved escalating CPU/ Memory usage.
